python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -615,37 +618,104 @@ def has_loadable_targets(self) -> bool:
+        """
         targets = []
         for target in self._targets:
             if isinstance(target, DirectTarget):
                 targets.append(target.load())
         return targets
 
-    def check_valid_run_requests(self, run_requests: Sequence[RunRequest]):
+    def resolve_run_requests(
+        self,
+        run_requests: Sequence[RunRequest],
+        context: SensorEvaluationContext,
+        asset_selection: Optional[AssetSelection],
+    ) -> Sequence[RunRequest]:
+        def _get_repo_job_by_name(context: SensorEvaluationContext, job_name: str) -> JobDefinition:
+            if context.repository_def is None:
+                raise DagsterInvariantViolationError(
+                    "Must provide repository def to build_sensor_context when yielding partitioned"
+                    " run requests"
+                )
+            return context.repository_def.get_job(job_name)
+
         has_multiple_targets = len(self._targets) > 1
         target_names = [target.pipeline_name for target in self._targets]
 
-        if run_requests and not self._targets and not self._asset_selection:
+        if run_requests and len(self._targets) == 0 and not self._asset_selection:
             raise Exception(
                 f"Error in sensor {self._name}: Sensor evaluation function returned a RunRequest "
                 "for a sensor lacking a specified target (job_name, job, or jobs). Targets "
                 "can be specified by providing job, jobs, or job_name to the @sensor "
                 "decorator."
             )
 
+        # Run requests may contain an invalid target, or a partition key that does not exist.
+        # We will resolve these run requests, applying the target and partition config/tags.
+        resolved_run_requests = []
+
         for run_request in run_requests:
             if run_request.job_name is None and has_multiple_targets:
                 raise Exception(
-                    f"Error in sensor {self._name}: Sensor returned a RunRequest that did not "
-                    f"specify job_name for the requested run. Expected one of: {target_names}"
+                    f"Error in sensor {self._name}: Sensor returned a RunRequest that did not"
+                    " specify job_name for the requested run. Expected one of:"
+                    f" {target_names}"
                 )
             elif run_request.job_name and run_request.job_name not in target_names:
                 raise Exception(
                     f"Error in sensor {self._name}: Sensor returned a RunRequest with job_name "
                     f"{run_request.job_name}. Expected one of: {target_names}"
                 )
 
+            if run_request.partition_key and not run_request.has_resolved_partition_run_request():
+                selected_job = _get_repo_job_by_name(
+                    context, run_request.job_name if run_request.job_name else target_names[0]
```

Comment:
Does this handle the case where `@sensor` isn't provided any job names and is instead just provided an asset selection, and there are multiple asset jobs?

Reply:
ah good catch--ive updated the PR to handle this